src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -16822,7 +16822,814 @@ public class Helper
+            comp.VerifyDiagnostics(
+                // (7,30): error CS9504: The type 'T' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'T' in the generic type or method 'Activator.CreateInstance<T>()'
+                //         _ = System.Activator.CreateInstance<T>();
+                Diagnostic(ErrorCode.ERR_NotRefStructConstraintNotSatisfied, "CreateInstance<T>").WithArguments("System.Activator.CreateInstance<T>()", "T", "T").WithLocation(7, 30),
+                // (12,30): error CS9504: The type 'S' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'T' in the generic type or method 'Activator.CreateInstance<T>()'
+                //         _ = System.Activator.CreateInstance<S>();
+                Diagnostic(ErrorCode.ERR_NotRefStructConstraintNotSatisfied, "CreateInstance<S>").WithArguments("System.Activator.CreateInstance<T>()", "T", "S").WithLocation(12, 30)
+                );
+        }
+
+        [Fact]
+        public void Field()
+        {
+            var src = @"
+#pragma warning disable CS0649 // Field is never assigned to, and will always have its default value
+
+class C<T>
+    where T : allows ref struct
+{
+    public T P1;
+    public S P2;
+}
+
+ref struct S1<T>
+    where T : allows ref struct
+{
+    public static T P3;
+    public static S P4;
+}
+
+ref struct S2<T>
+    where T : allows ref struct
+{
+    public T P5;
+    public S P6;
+}
+
+ref struct S
+{
+}
+";
+
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (7,12): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     public T P1;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(7, 12),
+                // (8,12): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public S P2;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(8, 12),
+                // (14,19): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     public static T P3;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(14, 19),
+                // (15,19): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public static S P4;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(15, 19)
+                );
+        }
+
+        [Fact]
+        public void AutoProperty()
```

Comment:
Consider testing similar scenario with synthesized record properties

Reply:
> Consider testing similar scenario with synthesized record properties

`RestrictedTypesInRecords` will be added in the next PR